### ISSUE TITLE
adamo: isometry_gradient can return a non-finite gradient

### ISSUE THREAD

**Issue by Svector-anu** [SlopDotCash/asi#1794]
`isometry_gradient` in `alberta_framework/core/adamo.py` (the closed-form gradient of `isometry_penalty`, paper equation 16, `arXiv:2606.09762v1`) has no finite-output guard, unlike its sibling `isometry_penalty` which already validates its own output.

Reproduction (current tree):

```text
weight = jnp.full((2, 2), jnp.finfo(jnp.float32).max, dtype=jnp.float32)
isometry_gradient(weight) -> [[inf inf] [inf inf]]
```

A weight matrix near the float32 max produces an infinite gradient that is returned silently instead of failing closed. This gradient feeds directly into `AdamO`'s isometry regularization step (`isometry_step = isometry_step_size * isometry_strength * isometry_gradient(parameter)`) when `regularize=True` and `isometry_strength != 0.0`.

`AdamO`/`isometry_gradient` are not re-exported from `alberta_framework.__init__` or `alberta_framework.core.__init__`, so reachability is module-level rather than package-root, but the caller is a real, non-test optimizer step method in the same module, not dead code.

Fix incoming in a linked PR: apply the same tracer-aware finite check `isometry_penalty` already uses — raise `ValueError` under eager execution, return `NaN` under `jax.jit`. No change to the gradient formula itself.

**Comment by lalalune** [comment]
Resolved on main by #1874. The consolidated implementation preserves the intended contract (including maintained package tooling instead of rewriting historical outputs where applicable), and post-merge main CI run 32201150336 is fully green.
